network-config-analyzer/PortSet.py

```diff
@@ -99,7 +99,17 @@ def __init__(self, source_ports=PortSet(), dest_ports=PortSet(), methods=None, p
         :param PortSet source_ports: The set of source ports (as a set of intervals/ranges)
         :param PortSet dest_ports: The set of target ports (as a set of intervals/ranges)
         """
-        self.cubes_set = CanonicalHyperCubeSet(TcpProperties.dimensions_list)
+        super().__init__(TcpProperties.dimensions_list)
+
+        self.named_ports = {}  # a mapping from dst named port (String) to src ports interval set
+        self.excluded_named_ports = {}  # a mapping from dst named port (String) to src ports interval set
+        # assuming named ports are only in dest, not src
+        all_ports = PortSet.all_ports_interval.copy()
+        for port_name in dest_ports.named_ports:
+            self.named_ports[port_name] = source_ports.port_set
+        for port_name in dest_ports.excluded_named_ports:
+            # self.excluded_named_ports[port_name] = all_ports - source_ports.port_set
+            self.excluded_named_ports[port_name] = all_ports
```

Comment:
should excluded named ports be mapped to source_ports as well? 
Can we simplify the named ports representation? (avoid the mapping to source ports?)

Reply:
We cannot avoid keeping mapping of named ports to source ports, since in case of dst ports consisting of named ports only (no real numbers), the cube is not added (since dim_val for dst_ports will be empty in the cube):
From TcpLikeProperties.__init__
            has_empty_dim_value = False
            for dim_val in cube:
                if not dim_val:
                    **has_empty_dim_value** = True
                    break
            if not **has_empty_dim_value**:
                self.add_cube(cube, active_dims)

Reply:
Discussed this with Ziv and he agreed that the most correct (and ugly :) way is to keep named_ports and excluded_named_ports as they are today